Include/internal/pycore_unionobject.h

```diff
@@ -8,9 +8,13 @@ extern "C" {
 #  error "this header requires Py_BUILD_CORE define"
 #endif
 
-PyAPI_FUNC(PyObject *) _Py_Union(PyObject *args);
-PyAPI_DATA(PyTypeObject) _Py_UnionType;
-PyAPI_FUNC(PyObject *) _Py_union_type_or(PyObject* self, PyObject* param);
+PyAPI_DATA(PyTypeObject) _PyUnion_Type;
+#define _PyUnion_Check(op) Py_IS_TYPE(op, &_PyUnion_Type)
```

Comment:
If we change it in future, the code will still use `_PyUnion_Check` instead of `_PyUnion_CheckExact`. So it is better to name it `_PyUnion_Check` now.